tests/integration/test_task_events_mgr.py

```diff
@@ -99,3 +101,39 @@ async def test__insert_task_job(flow, one_conf, scheduler, start, validate):
             i.submit_num for i
             in schd.data_store_mgr.added['jobs'].values()
         ] == [1, 2]
+
+
+async def test__always_insert_task_job(
+    flow, one_conf, scheduler, start, validate
+):
+    """Insert Task Job _Always_ inserts a task into the data store.
+
+    Bug https://github.com/cylc/cylc-flow/issues/6172 was caused
```

Comment:
We should be able to go a bit further with this integration test and probe the code pathways which result in these submit-fails to ensure the use case behaves correctly.

E.G, I tested with this:

```ini
# global.cylc

[platforms]    
    [[broken1]]    
        hosts = no-such-host-1
        install target = localhost
    [[broken2]]    
        hosts = no-such-host-2
        install target = localhost
         
[platform groups]    
    [[broken]]    
        platforms = broken1  
```

And this:

```ini
# flow.cylc

[scheduling]
    [[graph]]
        R1 = broken & broken2

[runtime]
    [[root]]
        submission retry delays = PT10M
    [[broken]]
        platform = broken
    [[broken2]]
        platform = broken2  # note broken2 not broken1 to avoid sharing hosts (opens the door to interactions via bad-hosts)
```

I.E, something along the lines of

```python
# untested

async with start(schd):
    for itask in schd.task_job_mgr.submit_task_jobs(
        self.workflow,
        self.pool.get_tasks(),
        self.server.curve_auth,
        self.server.client_pub_key_dir,
        is_simulation=False
    ):
        # check the task pool status
        assert itask.state.status == TASK_STATUS_WAITING
        assert itask.state.outputs.is_completed(TASK_OUTPUT_SUBMIT_FAILED)

        # check the data store status (see tests/integration/test_data_store_mgr.py for inspiration)
        await schd.update_data_store()
        assert {tp.name: tp.jobs[0].state for tp in schd.data_store_mgr.updated[TASK_PROXIES]}
```

Reply:
> assert itask.state.outputs.is_completed(TASK_OUTPUT_SUBMIT_FAILED)

Would you really expect this to be true if the task is waiting on retries?

Reply:
Ach, no, can remove that line.

Reply:
The main thing is to test the job statuses in the store.